app/Controllers/CtrlProduct.php

```diff
@@ -86,20 +84,34 @@ public function viewProductEdit($productId)
     public function createProduct()
     {
         try {
-            $productData                         = $this->request->getPost();
-            $productData['productTechnicalInfo'] = json_encode($productData['productTechnicalInfo']);
-
-            $validator = new ProductValidation();
-            $fileRules = $this->fileConfig->getCollectionFileValidationRules();
+            $productData = $this->request->getPost();
+            $validator   = new ProductValidation();
+            $fileRules   = $this->fileConfig->getCollectionFileValidationRules();
+
+            $keyValidation   = [];
+            $valueValidation = [];
+
+            foreach (array_keys($productData['keys'] ?? []) as $index) {
+                $keyValidation['rules']["keys.{$index}"]        = 'required';
+                $keyValidation['messages']["keys.{$index}"]     = ['required' => 'Campo obligatorio'];
+                $valueValidation['rules']["values.{$index}"]    = 'required';
+                $valueValidation['messages']["values.{$index}"] = ['required' => 'Campo obligatorio'];
+            }
```

Comment:
Hey @Mauriciocr207 please create a function for this you are duplicating code.

Lines 91 to 104 are the same as  Lines 146 and 163
```suggestion
           createProductDataKeysValidations($productData['keys'], $validator)
```

Reply:
Ready this changes